### PR TITLE
Ansible playbooks to provision and deploy

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

File: ansible/ACI/aci_create.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        name: "{{ resource_group }}"
+        location: "{{ location }}"
+    
+    - name: Create Azure Container Registry
+      azure_rm_containerregistry:
+        resource_group: "{{ resource_group }}"
+        name: "{{ registry_name}}"
+        admin_user_enabled: True
+        sku: Standard
+      register: acr_result
```
Comment (yungezz) [inline ansible/ACI/aci_create.yml]: no need to register output if not use it in same playbook

File: ansible/ACI/aci_deploy.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# Description
+# ===========
+# This playbook build the sample application into a docker image, push the image into ACR and create ACI from the image.
```
Comment (yungezz) [inline ansible/ACI/aci_deploy.yml]: do you want to include dependencies in comments to let user running docker before run this playbook

File: ansible/ACI/aci_create.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Description
+# ===========
+# This playbook create an Azure Container Registry.
```
Comment (yungezz) [inline ansible/ACI/aci_create.yml]: the file name is acr_create.yml right?

File: ansible/AKS/aks_create.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+     linux_profile:
+         admin_username: azureuser
+         ssh_key: "{{ SSH_PUBLIC_KEY }}"
+     agent_pool_profiles:
+         - name: default
+           count: 1
+           vm_size: Standard_DS1_v2
+    register: aks
+
+  - name: Create Azure Container Registry
```
Comment (yungezz) [inline ansible/AKS/aks_create.yml]: if i understand correctly, this is seperate scenario, one for aci, one for aks.  aks  creation seems in one playbook.  aci is seperated as acr_create.yml, aci_create.yml. why not merge acr_create.yml into aci_create.yml

File: ansible/VMSS/vmss_create.yml [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        allocation_method: Static
+        name: "{{ vmss_name }}"
+    - name: Create Network Security Group that allows SSH
+      azure_rm_securitygroup:
+        resource_group: "{{ resource_group }}"
+        name: "{{ vmss_name }}"
+        rules:
+          - name: SSH
+            protocol: Tcp
+            destination_port_range: 22
```
Comment (yungezz) [inline ansible/VMSS/vmss_create.yml]: why need open 22

File: ansible/ACI/aci_create.yml (hunk 1 above)
Comment (xiangyan99) [reply ansible/ACI/aci_create.yml]: Good catch. I will remove the code.

File: ansible/ACI/aci_deploy.yml (hunk 2 above)
Comment (xiangyan99) [reply ansible/ACI/aci_deploy.yml]: Added.

File: ansible/ACI/aci_create.yml (hunk 3 above)
Comment (xiangyan99) [reply ansible/ACI/aci_create.yml]: No. It is aci_create.yml. The naming convention is <scenarioname>_<action>. This is for the ACI scenario.

File: ansible/AKS/aks_create.yml (hunk 4 above)
Comment (xiangyan99) [reply ansible/AKS/aks_create.yml]: You are right. I try to make every e2e self-contained and avoid cross-references. Customers can copy the folder of a scenario and use run it directly.

File: ansible/VMSS/vmss_create.yml (hunk 5 above)
Comment (xiangyan99) [reply ansible/VMSS/vmss_create.yml]: After the VMSS provisioning, vmss_deploy need to ssh to those VMs to do the deployment which requires port 22 to be opened.

File: ansible/AKS/aks_vars.yml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+---
+resource_group: tododemo_aks
+resource_name: tododemoaks
+location: eastus
+AZURE_CLIENT_ID: your_azure_client_id
+AZURE_SECRET: your_azure_secret
```
Comment (yungezz) [inline ansible/AKS/aks_vars.yml]: suggest to get credentials from environment variables

**Pull request merged**